src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocParagraphCheck.java

```diff
@@ -189,6 +214,39 @@ else if (newLine == null || tag.getLineNumber() - newLine.getLineNumber() != 1)
         }
     }
 
+    /**
+     * Determines whether or not the paragraph tag is followed by block tag.
+     *
+     * @param tag html tag.
+     * @return block tag if the paragraph tag is followed by block tag.
+     */
+    private static String checkIfFollowedByBlockTag(DetailNode tag) {
+        String blockTag = null;
+        DetailNode htmlElement = JavadocUtil.getNextSibling(tag);
+
+        while (htmlElement != null && htmlElement.getType() != JavadocTokenTypes.HTML_ELEMENT) {
+            if (htmlElement.getType() == JavadocTokenTypes.TEXT
+                    && !CommonUtil.isBlank(htmlElement.getText())) {
+                htmlElement = null;
+                break;
+            }
+            htmlElement = JavadocUtil.getNextSibling(htmlElement);
+        }
```

Comment:
please move this to function findFirstHtmlElementAfter(tag)

Reply:
done